src/TreeHouse/Queue/Amqp/Driver/Amqp/Queue.php

```diff
@@ -0,0 +1,287 @@
+<?php
+
+namespace TreeHouse\Queue\Amqp\Driver\Amqp;
+
+use TreeHouse\Queue\Amqp\ChannelInterface;
+use TreeHouse\Queue\Amqp\QueueInterface;
+
+class Queue implements QueueInterface
+{
+    /**
+     * @var \AMQPQueue
+     */
+    protected $delegate;
+
+    /**
+     * @var ChannelInterface
+     */
+    protected $channel;
+
+    /**
+     * @var array<integer, integer>
+     */
+    protected static $flagMap = [
+        self::NOPARAM => AMQP_NOPARAM,
+        self::DURABLE => AMQP_DURABLE,
+        self::PASSIVE => AMQP_PASSIVE,
+        self::EXCLUSIVE => AMQP_EXCLUSIVE,
+        self::AUTODELETE => AMQP_AUTODELETE,
+        self::MULTIPLE => AMQP_MULTIPLE,
+        self::AUTOACK => AMQP_AUTOACK,
+        self::REQUEUE => AMQP_REQUEUE,
+        self::IFUNUSED => AMQP_IFUNUSED,
+        self::IFEMPTY => AMQP_IFEMPTY,
+    ];
+
+    /**
+     * @param \AMQPQueue       $delegate
+     * @param ChannelInterface $channel
+     */
+    public function __construct(\AMQPQueue $delegate, ChannelInterface &$channel)
+    {
+        $this->delegate = $delegate;
+        $this->channel = &$channel;
+    }
```

Comment:
Can you clarify why it's necessary to access the channel as a reference? Being an object, shouldn't it already have the same behaviour?